test/statetest/statetest_runner.cpp

```diff
@@ -19,25 +19,28 @@ void run_state_test(const StateTransitionTest& test, evmc::VM& vm)
             SCOPED_TRACE(std::string{evmc::to_string(rev)} + '/' + std::to_string(case_index));
             // if (rev != EVMC_FRONTIER)
             //     continue;
-            // if (case_index != 3)
+            // if (case_index != 0)
             //     continue;
 
             const auto& expected = cases[case_index];
             const auto tx = test.multi_tx.get(expected.indexes);
             auto state = test.pre_state;
 
-            // Validate deployed EOF containers before running state test.
+            // Validate deployed EOF containers before running state test since Cancun.
             // Any invalid EOF in state make the test incorrect.
-            for (const auto& [addr, acc] : state.get_accounts())
+            if (rev >= EVMC_CANCUN)
```

Comment:
Maybe this is good for now, but not sure long-term. We generally don't want garbage EOF in the state also before Cancun. And the validation function is also over-fitting by mapping Cancun to EOFv1.

@axic @gumb0 ?

Reply:
Before Cancun check that code doesn't start with `EF00`